File: backend/src/controllers/AuthController.ts
import { Request, Response } from 'express';
import { AuthService } from '../services/AuthService';
import { UsuarioLogin } from '../types/Usuario';

export class AuthController {
  static async login(req: Request, res: Response) {
    try {
      const credentials: UsuarioLogin = req.body;
      const result = await AuthService.login(credentials);
      res.json(result);
    } catch (error) {
      res.status(401).json({ message: error instanceof Error ? error.message : 'Erro ao fazer login' });
    }
  }

  static async register(req: Request, res: Response) {
    try {
      const userData = req.body;
      const newUser = await AuthService.register(userData);
      res.status(201).json(newUser);
    } catch (error) {
      res.status(400).json({ message: error instanceof Error ? error.message : 'Erro ao registrar usuário' });
    }
  }

  static async logout(req: Request, res: Response) {
    res.json({ message: 'Logout realizado com sucesso' });
  }

  static async me(req: Request, res: Response) {
    try {
      const token = req.headers.authorization?.split(' ')[1];
      if (!token) {
        return res.status(401).json({ message: 'Token não fornecido' });
      }

      const user = await AuthService.verifyToken(token);
      res.json(user);
    } catch (error) {
      res.status(401).json({ message: error instanceof Error ? error.message : 'Token inválido' });
    }
  }
} 

File: backend_dotnet/src/ViberLounge.API/Controllers/AuthController.cs
using Microsoft.AspNetCore.Mvc;
using ViberLounge.Domain.Entities;
using ViberLounge.Application.DTOs.User;
using Microsoft.AspNetCore.Authorization;

namespace ViberLounge.API.Controllers;

[ApiController]
[Route("[controller]")]
[Produces("application/json")]
public class AuthController : ControllerBase
{
    private readonly IAuthService _authService;

    public AuthController(IAuthService authService)
    {
        _authService = authService;
    }
    
    /// <summary>
    /// Realiza o login do usuário
    /// </summary>
    /// <param name="request">Dados de login</param>
    /// <returns>Token de autenticação</returns>
    /// <response code="200">Retorna o token de autenticação</response>
    /// <response code="401">Se as credenciais forem inválidas</response>
    [HttpPost("login")]
    [AllowAnonymous]
    [ProducesResponseType(typeof(LoginResponse), StatusCodes.Status200OK)]
    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
    public async Task<IActionResult> Login([FromBody] LoginRequest request)
    {
        try
        {
            var result = await _authService.LoginAsync(request);
            return Ok(result);
        }
        catch (UnauthorizedAccessException ex)
        {
            return Unauthorized(new { message = ex.Message });
        }
    }

    /// <summary>
    /// Registra um novo usuário
    /// </summary>
    /// <param name="request">Dados do novo usuário</param>
    /// <returns>Confirmação de registro</returns>
    /// <response code="201">Usuário registrado com sucesso</response>
    /// <response code="400">Se os dados do usuário forem inválidos</response>
    [HttpPost("register")]
    [AllowAnonymous]
    [ProducesResponseType(StatusCodes.Status201Created)]
    [ProducesResponseType(StatusCodes.Status400BadRequest)]
    public async Task<IActionResult> Register([FromBody] RegisterRequest request)
    {
        try
        {
            Usuario? newUser = await _authService.RegisterAsync(request);
            return Created(string.Empty, new { message = "Usuário cadastrado com sucesso" });
        }
        catch (Exception ex)
        {
            return BadRequest(new { message = ex.Message });
        }
    }
}